packages/rpc/lib/src/utils/unix_rpc_client.dart

```diff
@@ -17,12 +18,17 @@ class UnixRPCClient extends ServerProxyBase {
     var socket = await Socket.connect(address, 0);
     socket.add(utf8.encode(package));
 
-    socket.listen((event) {
-      LogManager.getInstance.debug('Event received is ${utf8.decode(event)}');
-      completer.complete(utf8.decode(event));
-    },
+    socket.listen(
+        (event) {
+          LogManager.getInstance
+              .debug('Event received is ${utf8.decode(event)}');
+          completer.complete(utf8.decode(event));
+        },
         onDone: () => LogManager.getInstance.debug('End JSON RPC Stream'),
-        onError: (err) => LogManager.getInstance.error(err));
+        onError: (err) {
+          LogManager.getInstance.error(err);
```

Comment:
```suggestion
          LogManager.getInstance.error(err);
```

This should be a json map, with the following form

{ 
   "error": { 
          "message": "",
          "code": "",
          "data": ""
     }

as defined in https://www.jsonrpc.org/specification#error_object
}

We can make a better exception, by adding error code and data also inside the class